meta-lmp-base/recipes-bsp/u-boot/u-boot-ostree-scr-fit/boot-common.cmd.in

```diff
@@ -39,7 +39,12 @@ else
 	if test -z "${upgrade_available}"; then setenv upgrade_available 0; setenv envsave 1; fi
 	if test -z "${bootupgrade_available}"; then setenv bootupgrade_available 0; setenv envsave 1; fi
 	# Call saveenv if not yet set (e.g. first boot after clean flash)
-	if test "${envsave}" = "1"; then setenv envsave; run saveenv_mmc; fi
+	if test "${envsave}" = "1"; then
+		# Also initialize bootfirmware_version to what is currently set at the rootfs
+		run bootcmd_bootenv
```

Comment:
why not to do the same as above? (so it aligns with all previous checks):
```
if test -z "${bootfirmware_version}"; then run bootcmd_bootenv; setenv envsave 1; fi
```

Reply:
Wasn't sure if this would be something we wanted to check on every boot, but thinking again I don't think it would cause any issue.

Will update with your suggestion.

Reply:
Done.